tests/aws/services/sns/test_sns.py

```diff
@@ -833,6 +816,39 @@ def subscribe_queue_to_topic(attributes: dict = None) -> dict:
             )
         snapshot.match("subscribe-diff-attributes", e.value.response)
 
+    @markers.aws.validated
+    def test_unsubscribe_idempotency(
+        self, sns_create_topic, sqs_create_queue, sns_create_sqs_subscription, snapshot, aws_client
+    ):
+        topic_name = f"topic-{short_uid()}"
+        queue_name = f"queue-{short_uid()}"
+        topic_arn = sns_create_topic(Name=topic_name)["TopicArn"]
+        queue_url = sqs_create_queue(QueueName=queue_name)
+        subscription = sns_create_sqs_subscription(topic_arn=topic_arn, queue_url=queue_url)
+        sub_arn = subscription["SubscriptionArn"]
+
+        unsubscribe_1 = aws_client.sns.unsubscribe(SubscriptionArn=sub_arn)
+        snapshot.match("unsubscribe-1", unsubscribe_1)
+        unsubscribe_2 = aws_client.sns.unsubscribe(SubscriptionArn=sub_arn)
+        snapshot.match("unsubscribe-2", unsubscribe_2)
```

Comment:
This test here seems to be valuable, but it I'm missing a test which actually tests the changes in the provider.
I can see that the positive case (the subscriptions are enabled and working) are tested in multiple tests here, but I think it would be great to also have a test checking that the logs are not written in case the logs are not explicitly enabled?

Reply:
True yes, that's a very good point! I just need to decide a fair value to estimate that the logs will never arrive and that we can assert the test is successful without adding a 30s test to the pipeline. But that would be very valuable to have!

Reply:
I added the test now, with some snapshot asserting that the first message wasn't logged and that the second message was properly logged, matching `messageId` to the events. Thanks for the review!